app/routes.py

```diff
@@ -1,2 +1,275 @@
+def specific_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return make_response(), 404
+    
+    elif request.method == "GET":
+
+        if task.goal == None:
+            return jsonify({
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.task_completed()
+                }
+            }), 200
+
+        else:
+            return jsonify({
+                "task": {
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.task_completed()
+                }
+            }), 200
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+        db.session.commit()
+
+        return jsonify({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.task_completed()
+            }
+        }), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify({"details": f'Task {task.task_id} "{task.title}" successfully deleted'}), 200 
+
+def slack_message_bot(task):
+
+    params = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.task_id}"
+    }
+
+    headers = {
+        "Authorization": f"Bearer {os.environ.get('SLACK_API_KEY')}"
+    }
+
+    requests.post("https://slack.com/api/chat.postMessage", data=params, headers=headers)
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    
+    if task == None:
+        return make_response(), 404
+    
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+
+    slack_message_bot(task)
+
+    return jsonify({
+        "task": {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.task_completed()
+        }
+        }), 200
+    
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task == None:
+        return make_response(), 404
+    
+    task.completed_at = None
+    db.session.commit()
+
+    return jsonify({
+            "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.task_completed()
+            }
+            }), 200
```

Comment:
Sometimes your code uses `make_response` and sometimes it doesn't. Consistently using it (or not) can add readability to your code.